lib/prefixed_ids.rb

```diff
@@ -123,4 +124,22 @@ def to_param
       _prefix_id.encode(id)
     end
   end
+
+  module Exists
+    extend ActiveSupport::Concern
+
+    class_methods do
+      def exists?(id)
+        return super(id) unless _prefix_id.present?
+
+        # `exists?` also accept conditions (e.g. `exists?(id: 1)`). These include, for example, Hashes and Arrays.
+        # In these cases, we know it can't be a prefixed ID so we just delegate to the original method.
+        return super(id) unless id.is_a?(String)
+
+        prefixed_id = _prefix_id.decode(id)
+
+        super(prefixed_id)
```

Comment:
The guards here feel a bit confusing to read. What if we simplified to something like this?
```suggestion
      def exists?(id)
        if _prefix_id.present? && id.is_a?(String)
          super(_prefix_id.decode(id))
        else
          super(id)
```

Reply:
yup, that makes sense. this is a leftover from before I realized `exists?` also accept other type of data as well, I just tossed it in there